### PR TITLE
Snow control plane reconcile phase

### PR DESCRIPTION
**Pull request by taneyland**
*Issue #, if available:*

*Description of changes:*
Add the cp reconcile phase for snow cluster controller. 

*Testing (if applicable):*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by mitalipaygude** [comment]
Can you add some description for this issue?

File: pkg/clusterapi/machine_health_check.go [hunk 1]
```diff
@@ -80,9 +80,9 @@ func machineHealthCheckForWorker(clusterSpec *cluster.Spec, workerNodeGroupConfi
 }
 
 // MachineHealthCheckObjects creates MachineHealthCheck resources for control plane and all the worker node groups.
-func MachineHealthCheckObjects(clusterSpec *cluster.Spec) []runtime.Object {
+func MachineHealthCheckObjects(clusterSpec *cluster.Spec) []client.Object {
```
Comment (g-gaston) [inline pkg/clusterapi/machine_health_check.go:83]: Do we need to change this?

File: pkg/providers/snow/reconciler/reconciler.go [hunk 2]
```diff
@@ -48,5 +49,25 @@ func (r *Reconciler) Reconcile(ctx context.Context, log logr.Logger, c *anywhere
...
+
+func (r *Reconciler) reconcileControlPlaneSpec(ctx context.Context, cluster *anywherev1.Cluster, specWithBundles *cluster.Spec, kubeClient *clientutil.KubeClient, log logr.Logger) (controller.Result, error) {
+	log.Info("Applying control plane spec", "name", cluster.Name)
+	objs, err := snow.ControlPlaneObjects(ctx, specWithBundles, kubeClient)
+	if err != nil {
+		return controller.Result{}, err
+	}
+	for _, obj := range objs {
+		key := client.ObjectKey{Name: obj.GetName(), Namespace: obj.GetNamespace()}
+		if err := r.client.Get(ctx, key, obj); err != nil {
```
Comment (g-gaston) [inline pkg/providers/snow/reconciler/reconciler.go:68]: I might be missing something, but what is this supposed to do? Shouldn't we be creating/updating the objects?

File: pkg/templater/yaml.go [hunk 3]
```diff
@@ -26,7 +26,7 @@ func AppendYamlResources(resources ...[]byte) []byte {
 	return b
 }
 
-func ObjectsToYaml(objs ...runtime.Object) ([]byte, error) {
+func ObjectsToYaml(objs ...client.Object) ([]byte, error) {
```
Comment (g-gaston) [inline pkg/templater/yaml.go:29]: Do we need to change this too?

**Review by g-gaston — APPROVED** [review]

**Pull request merged**